### PR TITLE
[INLONG-8805][Manager] Check tenant status before deleting tenant

### LINKED ISSUE #8805: [Feature][Manager] Check tenant status before deleting tenant
### Description

Check the following resources' status before deleting a tenant:
1. If all Groups are in stop status.
2. If all  StreamSource of these groups are in status 99.

If the above conditions are not satisfied, return false and indicate to the user which groups or sources are still running.

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by kamping**
[INLONG-8805][Manager] Check tenant status before deleting tenant

### Prepare a Pull Request
*[INLONG-8805][Manager] Check tenant status before deleting tenant*

- Title Example: [INLONG-XYZ][Component] Title of the pull request

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #8805

### Motivation

[INLONG-8805][Manager] Check tenant status before deleting tenant*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/tenant/InlongTenantServiceImpl.java [hunk 1]
```diff
@@ -156,6 +167,32 @@ public Boolean delete(String name) {
...
         log.info("begin to delete inlong tenant name={} by user={}", name, operator);
         InlongTenantEntity inlongTenantEntity = inlongTenantEntityMapper.selectByName(name);
+        // before deleting a tenant, check if all Groups of the tenant are in stop status
+        List<InlongGroupEntity> groupList = groupMapper.selectByTenant(name);
+        long notStopGroupCount =
+                groupList.stream().filter(
+                        group -> !GroupStatus.DELETED.getCode().equals(group.getStatus()) || group.getIsDeleted() == 0)
+                        .count();
+        if (notStopGroupCount > 0) {
+            String errMsg = String.format(
```
Comment (vernedeng) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/tenant/InlongTenantServiceImpl.java:177]: Please log out all groups that are not in stop or deleted status

Comment (kamping) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/tenant/InlongTenantServiceImpl.java:179]: finish

**Review by vernedeng — APPROVED** [review]

File: inlong-manager/manager-dao/src/main/resources/mappers/InlongGroupEntityMapper.xml [hunk 2]
```diff
@@ -553,4 +553,13 @@
             </if>
         </where>
     </delete>
+    <select id="selectAllGroupsByTenant" resultMap="BaseResultMap">
+        select
+        <include refid="Base_Column_List"/>
+        from inlong_group
+        <where>
+            tenant = #{tenant,jdbcType=VARCHAR}
```
Comment (fuweng11) [inline inlong-manager/manager-dao/src/main/resources/mappers/InlongGroupEntityMapper.xml:561]: and is_deleted = 0

Comment (kamping) [reply inlong-manager/manager-dao/src/main/resources/mappers/InlongGroupEntityMapper.xml:561]: modified

**Review by fuweng11 — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**